### PR TITLE
Add support for SET ACCESS METHOD in altering a distributed table

### PR DESCRIPTION
**Pull request by TsinghuaLucky912**
#6381 

Dear hackers, this problem seems to be solved with one line of code. I don’t know if I think it’s too simple.

```
postgres=# create table t1 (id int primary key);
CREATE TABLE
postgres=# set citus.shard_count to 2;
SET
postgres=# select create_distributed_table('t1', 'id');
 create_distributed_table 
--------------------------
 
(1 row)

postgres=# \d
            List of relations
 Schema |     Name     | Type  |  Owner   
--------+--------------+-------+----------
 public | citus_tables | view  | postgres
 public | t1           | table | postgres
(2 rows)

postgres=# set citus.override_table_visibility TO off;
SET
postgres=# \d+ t1
                                           Table "public.t1"
 Column |  Type   | Collation | Nullable | Default | Storage | Compression | Stats target | Description 
--------+---------+-----------+----------+---------+---------+-------------+--------------+-------------
 id     | integer |           | not null |         | plain   |             |              | 
Indexes:
    "t1_pkey" PRIMARY KEY, btree (id)
Access method: heap

postgres=# CREATE ACCESS METHOD heap2 TYPE TABLE HANDLER heap_tableam_handler;
CREATE ACCESS METHOD
postgres=# set citus.log_remote_commands=on;
SET
postgres=# 
postgres=# ALTER TABLE t1 SET ACCESS METHOD heap2;
NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(0, 2, '2022-11-28 06:08:03.196326-08');
DETAIL:  on server postgres@localhost:5432 connectionId: 1
NOTICE:  issuing SELECT worker_apply_shard_ddl_command (102014, 'public', 'ALTER TABLE t1 SET ACCESS METHOD heap2;')
DETAIL:  on server postgres@localhost:5432 connectionId: 1
NOTICE:  issuing SELECT worker_apply_shard_ddl_command (102015, 'public', 'ALTER TABLE t1 SET ACCESS METHOD heap2;')
DETAIL:  on server postgres@localhost:5432 connectionId: 1
NOTICE:  issuing PREPARE TRANSACTION 'citus_0_72994_2_1'
DETAIL:  on server postgres@localhost:5432 connectionId: 1
NOTICE:  issuing COMMIT PREPARED 'citus_0_72994_2_1'
DETAIL:  on server postgres@localhost:5432 connectionId: 1
ALTER TABLE
postgres=# \d
            List of relations
 Schema |     Name     | Type  |  Owner   
--------+--------------+-------+----------
 public | citus_tables | view  | postgres
 public | t1           | table | postgres
 public | t1_102014    | table | postgres
 public | t1_102015    | table | postgres
(4 rows)

postgres=# \d+ t1_102014
                                        Table "public.t1_102014"
 Column |  Type   | Collation | Nullable | Default | Storage | Compression | Stats target | Description 
--------+---------+-----------+----------+---------+---------+-------------+--------------+-------------
 id     | integer |           | not null |         | plain   |             |              | 
Indexes:
    "t1_pkey_102014" PRIMARY KEY, btree (id)
Access method: heap2

postgres=#
```

### CODE REVIEW

File: src/test/regress/expected/pg15.out [hunk 1]
```diff
@@ -1456,6 +1456,36 @@ NOTICE:  drop cascades to 2 other objects
...
+    key int primary key,
+    value int
+);
+SELECT create_distributed_table('mx_ddl_table2', 'key', 'hash', shard_count=> 4);
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+ALTER TABLE mx_ddl_table2 SET ACCESS METHOD heap2;
```
Comment (marcocitus) [inline src/test/regress/expected/pg15.out:1479]: this could probably just do `SET ACCESS METHOD citus_columnar`

**Review by marcocitus — APPROVED** [review]

**Comment by TsinghuaLucky912** [comment]
@marcocitus Hello, Excuse me. This change has the macro restriction of pg15. It should not affect pg13. Could it be that there is something wrong with the regression test? Can you run CI again?

**Comment by marcocitus** [comment]
> Hello, I don't know what happened. I tested the following regression test on my local environment and there is no error (pg13.0)

I think some flappy tests due to recent commits. We'll look into it.

**Comment by TsinghuaLucky912** [comment]
@marcocitus Thank you very much for your help. There is still a merge that needs to be reviewed. It has been stagnant for a long time. I don't know what happened. Can you help me take a look? #6519

**Pull request merged**